src/lib/OS/MemRegion.cc

```diff
@@ -186,25 +188,21 @@ int64_t MemRegion::removeVma(uint64_t addr, uint64_t length) {
     // overlapping region of the VMA.
     //          [--------Addr--------]
     //  [--------VMA --------)
-    else if (curr->overlaps(addr, length)) {
-      if (addr > curr->vmStart_ && endAddr > curr->vmEnd_) {
-        delsize += (curr->vmEnd_ - addr);
-        curr->trimRangeEnd(addr);
-        prev = curr;
+    else if (itr->overlaps(addr, length)) {
+      if (addr > itr->vmStart_ && endAddr > itr->vmEnd_) {
+        delsize += (itr->vmEnd_ - addr);
+        itr->trimRangeEnd(addr);
+        itr++;
       } else {
-        delsize += (endAddr - curr->vmStart_);
-        curr->trimRangeStart(endAddr);
+        delsize += (endAddr - itr->vmStart_);
+        itr->trimRangeStart(endAddr);
         break;
       }
```

Comment:
Could there not be a case where the given `addr` is contained within one VMA, and then `addr+length` is in another VMA? i.e. the region to delete overlaps two VMAs?

Reply:
Yeah, all those cases have been handled if you look at MemRegionTest.cc there are tests for the overlapping unmaps (with illustrations)

Reply:
Sorry, I miss-read the implementation